schema/python/schemaorgutils/validator.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import rdflib
+import uuid
+import os
+import schemaorgutils.utils.constants as constants
+import schemaorgutils.utils.utils as utils
+from pyshacl import validate
+from jinja2 import Environment, FileSystemLoader
+
+class SchemaValidator():
+    """The SchemaValidator validates the entities against a constraints graph and generate a html report for the validation result.
+
+    Attributes:
+        reports (dict(list(ResultRow))): A dictionary mapping list of all error results to an entity type.
+        _constraints_file (str): The path to constraints file containing shacl validations.
+        _report_file (str): The path to file where the output report has to be generated.
+        _position (int): The total number of entities validated.
+        _is_closed (bool): The status of validator.
+    """
+    
+    def __init__(self, constraints_file, report_file):
```

Comment:
Check if the underlying library can take an array of files. If yes, then update this param. If no, then it's fine.